prompttools/test/test_hugging_face_hub.py

```diff
@@ -0,0 +1,92 @@
+# Copyright (c) Hegel AI, Inc.
+# All rights reserved.
+#
+# This source code's license can be found in the
+# LICENSE file in the root directory of this source tree.
+
+from prompttools.experiment import HuggingFaceHubExperiment
+from prompttools.utils import similarity
+from typing import Dict, List
+
+
+def test_single_model() -> None:
```

Comment:
For unit testing the prompttools library, we have a DEBUG mode that uses mock LLM calls instead of hitting real APIs

https://github.com/hegelai/prompttools/blob/main/prompttools/experiment/openai_chat_experiment.py#L63-L64

I think it would be best to implement a mock HF function and run this test in debug mode. Otherwise it will require authentication and be flaky. We want to separate those issues from our unit tests. 

If you want to add an example of testing with the HF experiment (that is, test outside of DEBUG mode), it would be best to do so in `examples/prompttest`

Also, it might be beneficial to implement a test runner and harness for the huggingface experiment class. For reference, check these out 

* Test Runner: https://github.com/hegelai/prompttools/blob/main/prompttools/testing/runner/prompt_template_runner.py
* Harness: https://github.com/hegelai/prompttools/blob/main/prompttools/harness/prompt_template_harness.py

The idea behind a test runner is that it runs unit tests for a set of prompts given an evaluation function and some parameters. A harness is an abstraction built on top of an experiment that helps experiment over specific inputs, e.g. prompt templates or system prompts. Check out this example: 

https://github.com/hegelai/prompttools/blob/main/examples/prompttests/example.py